src/core/analytics.js

```diff
@@ -392,15 +392,26 @@ class Analytics {
                 let transformedEvents;
                 if (transformationServerAccess) {
                   // filter the transformed event for that destination
-                  transformedEvents = transformedPayload.find(
-                    (e) => e.id === intg.destinationId,
-                  ).payload;
+                  transformedEvents = transformedPayload.find((e) => e.id === intg.destinationId);
+                  if (!transformedEvents) {
+                    logger.error(
+                      `[DMT]::[Destination:${intg.name}]:: Transformed data not available`,
+                    );
+                  }
+                  /**
+                   * Filter the successful transformed event
+                   */
+                  transformedEvents = transformedEvents?.payload.filter(
+                    (tEvent) => tEvent.status === '200',
+                  );
                 } else {
                   transformedEvents = transformedPayload;
                 }
                 // send transformed event to destination
-                transformedEvents.forEach((tEvent) => {
-                  this.sendDataToDestination(intg, { message: tEvent.event }, methodName);
+                transformedEvents?.forEach((tEvent) => {
```

Comment:
should we not sent only 200 status with not null | undefined value for event property?

Reply:
If the rudder event is null , there is no point sending it to device mode destination